samples/fleet_provisioning/fleet_provisioning/main.cpp

```diff
@@ -21,87 +27,82 @@
 #include <aws/iotidentity/RegisterThingResponse.h>
 #include <aws/iotidentity/RegisterThingSubscriptionRequest.h>
 
-#include <algorithm>
-#include <chrono>
-#include <condition_variable>
 #include <fstream>
-#include <iostream>
-#include <mutex>
-#include <sstream>
-#include <streambuf>
-#include <string>
-#include <thread>
 
 #include "../../utils/CommandLineUtils.h"
 
 using namespace Aws::Crt;
 using namespace Aws::Iotidentity;
-using namespace std::this_thread; // sleep_for, sleep_until
-using namespace std::chrono;      // nanoseconds, system_clock, seconds
 
-static void sleep(int sleeptime)
+static String getFileData(const String &fileName)
 {
-    std::cout << "Sleeping for " << sleeptime << " seconds..." << std::endl;
-    sleep_until(system_clock::now() + seconds(sleeptime));
-}
-
-static std::string getFileData(std::string const &fileName)
-{
-    std::ifstream ifs(fileName);
+    std::ifstream ifs(fileName.c_str());
     std::string str;
     getline(ifs, str, (char)ifs.eof());
-    return str;
+    return str.c_str();
 }
 
-int main(int argc, char *argv[])
+/**
+ * Auxiliary structure for holding data used by MQTT connection.
+ */
+struct ConnectionContext
 {
-    /************************ Setup ****************************/
+    std::promise<bool> connectionCompletedPromise;
+    std::promise<void> connectionClosedPromise;
+};
 
-    //  Do the global initialization for the API
-    ApiHandle apiHandle;
-    // Variables for the sample
-    String csrFile;
+/**
+ * Auxiliary structure for holding data used when creating a certificate.
+ */
+struct CreateCertificateContext
+{
+    std::promise<void> pubAckPromise;
+    std::promise<void> acceptedSubAckPromise;
+    std::promise<void> rejectedSubAckPromise;
+    std::promise<void> tokenReceivedPromise;
```

Comment:
Technically could probably merge token into the tokenReceivedPromise by changing it to std::promise<String> but not a big deal.

Reply:
That's a good improvement, I'll do it.